src/cephadm/cephadm

```diff
@@ -4608,6 +4609,336 @@ def command_install():
+                continue
+            for iface in os.listdir(nic_path):
+
+                mtu = rfiles([os.path.join(nic_path, iface, 'mtu')])
+                operstate = rfiles([os.path.join(nic_path, iface, 'operstate')])
+                try:
+                    speed = rfiles([os.path.join(nic_path, iface, 'speed')])
+                except OSError:
+                    # OSError is raised when the device doesn't support
+                    # the ethtool get_link_ksettings, so we'll use -1
+                    # to indicate unknown (-1 is also shown for NICs
+                    # that are not link-up)
+                    speed = "-1"
+
+                iftype = 'physical' if os.path.exists(os.path.join(nic_path, iface, 'device')) else 'logical'
+
+                self.interfaces[iface] = {
+                    "mtu": mtu,
+                    "operstate": operstate,
+                    "iftype": iftype,
+                    "speed": speed,
+                    "ipv4_address": get_ipv4_address(iface),
+                    "ipv6_address": get_ipv6_address(iface),
+                }
+
+    @property
+    def nic_count(self):
+        # type: () -> int
+        phys_devs = []
+        for iface in self.interfaces:
+            if self.interfaces[iface]["iftype"] == 'physical':
+                phys_devs.append(iface)
+        return len(phys_devs)
+
+    @property
+    def memory_kb(self):
+        # type: () -> int
+        out = self._memory.splitlines()
+        for line in out:
+            if line.startswith('MemTotal'):
+                _d = line.split()
+                return int(_d[1])
+        return 0
+
+    @property
+    def vendor(self):
+        # type: () -> str
+        return rfiles(HostFacts._dmi_path_list, "sys_vendor")
+
+    @property
+    def model(self):
+        # type: () -> str
+        family = rfiles(HostFacts._dmi_path_list, "product_family")
+        product = rfiles(HostFacts._dmi_path_list, "product_name")
+        if family == 'Unknown' and product:
+            return "{}".format(product)
+
+        return "{} ({})".format(family, product)
+
+    @property
+    def bios_version(self):
+        # type: () -> str
+        return rfiles(HostFacts._dmi_path_list, "bios_version")
+
+    @property
+    def bios_date(self):
+        # type: () -> str
+        return rfiles(HostFacts._dmi_path_list, "bios_date")
+
+    def dump(self, format_type='json'):
+        # type: (str) -> str
+        return self._to_json()
+
+    def _to_json(self):
+        # type: () -> str
+        data = {k: getattr(self, k) for k in dir(self)
+                if not k.startswith('_') and
+                isinstance(getattr(self, k), 
+                           (float, int, str, list, dict, tuple))
+        }
+        return json.dumps(data, indent=2, sort_keys=True)
+
+
+##################################
+
+def command_gather_facts():
+    host = HostFacts()
+    print(host.dump(args.fact_format))
```

Comment:
Mind adding this to https://github.com/ceph/ceph/blob/master/qa/workunits/cephadm/test_cephadm.sh  ?

Reply:
ACK